src/libcommonserver/io/iohelper_win.cpp

```diff
@@ -329,126 +331,286 @@ bool IoHelper::checkIfFileIsDehydrated(const SyncPath &itemPath, bool &isDehydra
+
+    auto pathwPtr = std::make_unique<WCHAR[]>(pathLen + 1);
+    pathWstr.copy(pathwPtr.get(), pathLen);
+
+    LPCWSTR pathw_c = pathWstr.c_str();
+    LPWSTR pathw = pathwPtr.get();
+    pathw[pathLen] = L'\0';
+
+    DWORD valueReturned = GetNamedSecurityInfo(pathw_c, SE_FILE_OBJECT, DACL_SECURITY_INFORMATION, nullptr, nullptr, &pACLold,
+                                               nullptr, &pSecurityDescriptor);
+    ioError = dWordError2ioError(valueReturned);
+    if (valueReturned != ERROR_SUCCESS) {
+        ioError = dWordError2ioError(valueReturned);
+        LOGW_WARN(logger, L"Error in GetNamedSecurityInfo: " << Utility::formatIoError(path, ioError).c_str()
+                                                             << L" | DWORD error: " << valueReturned);
+        LocalFree(pSecurityDescriptor);
+        LocalFree(pACLnew);
+        // pACLold is a pointer to the ACL in the security descriptor, so it should not be freed.
+        if (ioError == IoErrorNoSuchFileOrDirectory || ioError == IoErrorAccessDenied) {
+            return true;
+        }
+        return false;
+    }
+
+    valueReturned = SetEntriesInAcl(1, &explicitAccess, pACLold, &pACLnew);
+    ioError = dWordError2ioError(valueReturned);
+    if (valueReturned != ERROR_SUCCESS) {
+        ioError = dWordError2ioError(valueReturned);
+        LOGW_WARN(logger, L"Error in SetEntriesInAcl: " << Utility::formatIoError(path, ioError).c_str() << L" | DWORD error: "
+                                                        << valueReturned);
+        LocalFree(pSecurityDescriptor);
+        LocalFree(pACLnew);
+        // pACLold is a pointer to the ACL in the security descriptor, so it should not be freed.
+        if (ioError == IoErrorNoSuchFileOrDirectory || ioError == IoErrorAccessDenied) {
+            return true;
+        }
+        return false;
+    }
+
+    if (!IsValidAcl(pACLnew)) {
+        ioError = IoErrorUnknown;
+        LOGW_WARN(logger, L"Invalid new ACL: " << Utility::formatSyncPath(path).c_str());
+
+        LocalFree(pSecurityDescriptor);
+        LocalFree(pACLnew);
+        // pACLold is a pointer to the ACL in the security descriptor, so it should not be freed.
+        return false;
+    }
+
+    valueReturned = SetNamedSecurityInfo(pathw, SE_FILE_OBJECT, DACL_SECURITY_INFORMATION, nullptr, nullptr, pACLnew, nullptr);
+    ioError = dWordError2ioError(valueReturned);
+    if (valueReturned != ERROR_SUCCESS) {
+        ioError = dWordError2ioError(valueReturned);
+        LOGW_WARN(logger, L"Error in SetNamedSecurityInfo: " << Utility::formatIoError(path, ioError).c_str()
+                                                             << L" | DWORD error: " << valueReturned);
+        LocalFree(pSecurityDescriptor);
+        LocalFree(pACLnew);
+        // pACLold is a pointer to the ACL in the security descriptor, so it should not be freed.
+        if (ioError == IoErrorNoSuchFileOrDirectory || ioError == IoErrorAccessDenied) {
```

Comment:
This part is replace by return false, the caller then check with _isExpectedError. (commit:  https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/5b1ece6f3db1e3c4fb9b55758d5044286324abe4 & https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/a4fd7e36b99627986f1103db69d5d23a69f4a40c).

Reply:
My suggestion was to remove `return false` (just return the above boolean).

Reply:
In the first implementation, I used 
```
if (ioError == IoErrorNoSuchFileOrDirectory || ioError == IoErrorAccessDenied) {
    return true;
} 
... 
return false;

```

Because `setRightsWindowsApi` is not a IoHelper member and so could not access `_isExpectedError`.

To avoid duplication of `_isExpectedError` code and risk to break the code in case we modified `_isExpectedError `without modifying these conditions, `setRightsWindowsApi` now always return false if `ioError != IoErrorSuccess` and the caller check `_isExpectedError`.

Reply:
I had implemented this behavior for getRightsWindowsApi, as well as for the call to setRightsWindowsApi, but apparently neglected to do so in the setRightsWindowsApi implementation. Your suggestion helped me spot this, Thanks you!